app/models/post.rb

```diff
@@ -0,0 +1,16 @@
+class Post < ApplicationRecord
+  has_many :likes, foreign_key: 'posts_id'
+  has_many :comments, foreign_key: 'posts_id'
+  belongs_to :author, class_name: 'User', dependent: :destroy, foreign_key: 'author_id'
```

Comment:
- [ ] At that point using dependent destroy with belongs_to is not ok. Why? because you are destroying your user when you are deleting only one post. Try to use dependent destroy with has_many, not with belongs_to in this project. Happy coding!!!